src/components/Recovery.tsx

```diff
@@ -487,11 +487,11 @@ export default function Recovery() {
   const { push } = useHistory();
   const { enqueueSnackbar } = useSnackbar();
   const dispatch = useDispatch();
-  const { provider } = useEthereumProvider();
-  const [type, setType] = useState("Token");
-  const isNFT = type === "NFT";
   const [recoverySourceChain, setRecoverySourceChain] =
     useState<ChainId>(CHAIN_ID_SOLANA);
+  const { provider } = useEthereumProvider(recoverySourceChain);
+  const [type, setType] = useState("Token");
```

Comment:
Can we do this? So we make sure this `type` value is only one of those.
```suggestion
  const [type, setType] = useState<"Token" | "NFT">("Token");
```